### DISCUSSION TITLE
Inactive session data in `session` table in DB

### DISCUSSION THREAD

**Discussion by MaheshkumarSundaram** [neonexus/sails-react-bootstrap-webpack#91] [General]
Hi @neonexus,

Usually the users log on to the web app, their sessions are stored in the `session` table in DB. When they logout, the data are deleted accordingly. Operation is fine.

But when the users doesn't explicitly logout and comeback after a session timeout or any abrupt ending of the session, I believe the corresponding data doesn't get removed and is retained in DB. Mostly users don't explicitly logout and such inactive session data keeps accumulating.

Would you recommend any approach to effectively remedy this?

Thank you!

**Comment by neonexus** [comment]
So, overall, I wouldn't be too concerned about session data build-up. The storage size is almost nothing, even at 100k records. 

However, it is still good to clear the cobwebs from time to time.

I would look into an NPM CRON job handler (I'm pretty sure there was, maybe still is a Sails Hook). Then, once a day, clear out sessions that haven't been updated in the past 90 days (or however long); making sure to use the slowest time to do so.

**Reply by MaheshkumarSundaram** [reply]
I am not concerned about the storage size. It's just so many inactive records.

I'll look into the Sails hook if available or a CRON job. Thank you:relaxed:!

**Comment by neonexus** [comment]
Something recently came to mind, that I thought I built in, but hadn't... I used to make it standard practice to have an `expiresAt` dateTime on sessions, that the `isLoggedIn` policy would check. If the time expired, it would delete the session, then return a "not logged in" response. I also had it so the `/me` API call would update the expiry, so sessions wouldn't die pre-maturely.

I'm likely going to add this in to my next update.

**Reply by MaheshkumarSundaram** [reply]
This seems more regulated and proper. Thank you @neonexus.
